routes/index.js

```diff
@@ -313,150 +337,275 @@ router.post('/filter', function(req, res, next) {
+                            playlists: playlists,
+                            // messages: messages,
+                            // langsExist: langsExist,
+                        })
+                    })
+                })
+            } else {
+                res.render('search', {
+                    songs: songs2d,
+                    inLibrary: [],
+                    playlists: [],
+                    // messages: messages,
+                    // langsExist: langsExist,
+                })
+            }
+        }
+    }
+
+    if (searchString) {
+        async.waterfall([findOriginalSong], finalize)
+    } else {
+        res.render('search', {
+            songs: [],
+            inLibrary: [],
+            playlists: [],
+            // messages: messages,
+            // langsExist: langsExist,
+        })
+    }
+})
+
+router.get('/discover', function(req, res, next) {
+    var messages = req.flash();
+    var langsExist;
     var messages = req.flash()
     Song.find({
-            $text: {
-                $search: "\"" + tag + "\""
-            },
             private: false
         }, function(err, songs, count) {
             if (err) {
                 res.status(400).send('error getting song list ' + err)
             }
+            //to get the what languages we need to include in the dropdown for filtering feature
+            langsExist = _.uniq(songs.map((s) => s.lang))
             if (req.isAuthenticated()) {
                 User.findOne({
                     _id: req.user._id
                 }, function(err, user) {
                     if (err) next(err)
                     Playlist.find({
-                        owner: req.user._id
+                        owner: req.user._id,
+                        song: {
+                            $exists: false
+                        }
                     }, function(err, playlists) {
                         if (err) next(err)
-                        res.render('index', {
+                        res.render('songlist', {
                             songs: songs,
                             inLibrary: user.library,
                             playlists: playlists,
-                            messages: messages
+                            messages: messages,
+                            langsExist: langsExist,
+                            isLoggedIn: true
                         })
                     })
-
                 })
             } else {
-                res.render('index', {
+                res.render('songlist', {
                     songs: songs,
                     inLibrary: [],
                     playlists: [],
-                    messages: messages
+                    messages: messages,
+                    langsExist: langsExist,
+                    isLoggedIn: false
                 })
             }
         })
         .sort({
             title: 1
         })
+        .limit(10)
 })
 
+router.route('/song/:song_id')
```

Comment:
Since this file is getting long, consider breaking it up into several smaller files that you load into this one.
Certain routes can be grouped together into their own files for different features. Makes it easier to maintain and read the code.